### PR TITLE
Assorted sorbet.org/docs/ changes

### PR DESCRIPTION
**Pull request by jez**
### Commit summary

Recommend reviewing by commit.

- **docs/final.md: Shorten sidebar label** (e4be1a76b)

  This makes it fit onto oneline. This was previously our only sidebar 
  item that didn't fit on one line.

- **docs/procs.md: Follow up to changes in #1560** (4c220e27f)

  I was thinking about leaving these as feedback on the PR itself, but I 
  figured it'd be faster for me to just make them in a separate commit.

  cc @connorshea

- **docs/final.md: Use more words** (6d65479ca)

- **sidebars.json: Reorder pages** (4879f015c)

  This is updated to be more in line with the kinds of questions people 
  get. Also moving tuples and shapes to "Experimental Features" to 
  discourage their use a little more.

- **docs/abstract.md: Use ampersand in sidebar_label** (99c54ef8a)

- **docs/procs.md: Make sure the word "block" appears in the sidebar** (91e05109b)

  People usually know they're searching for blocks, not procs.

- **docs/tstruct.md: Show T::Struct more prominently** (1525d3b93)

- **docs/abstract.md: Assorted edits** (1d18e9c8d)

  Most importantly: softens the language about some things are only 
  implemented in the runtime, as now most of these errors are implemented 
  statically, with variance checks being the last big holdout.

- **docs/override-checking.md: Flesh it out a little** (ece21b5e9)

  Also invert it to say "mostly statically checked" instead of "mostly 
  runtime."

- **Add some "What's next?" sections to cross link** (a973d291d)

### Motivation

I'd been putting off a bunch of these for a while. I figured I'd make myself an
"update the docs" day given that I'm also adding a doc for `sealed!` (got that
"docs writing" state of mind).

### Test plan

There are too many modified changes to be worth taking screenshots of. To view
these change locally:

```
git pr checkout 1632
cd website
yarn start
```

I previewed these while I was editing them.

### CODE REVIEW

File: website/docs/procs.md [hunk 1]
```diff
@@ -17,105 +18,147 @@ At present, all parameters are assumed to be required positional parameters. We
...
   yield
 end
 ```
 
-This error can be fixed without modifying the method body to use `block.call`:
-It's possible to give the method a named block parameter like `&blk`, and
-`yield` will still work exactly as it did, invoking the block regardless of
-whether or not it was named. Once the block parameter has a name, it's in turn
-possible to associate that name with a type:
+This is valid Ruby, and Sorbet accpet this code too. Implicitly, Sorbet will
```
Comment (connorshea) [inline website/docs/procs.md]:
```suggestion
This is valid Ruby, and Sorbet will accept this code too. Implicitly, Sorbet will
```

File: website/docs/abstract.md [hunk 2]
```diff
@@ -20,6 +16,9 @@ Keep in mind:
 - Mix in a module (via `include` or `extend`) to declare that a class implements
   an interface.
 
+> **Note**: Most of the abstract and override checks are implemented statically,
+> but some are still only implemented at runtime, most notably variance checks.
```
Comment (DarkDimius) [inline website/docs/abstract.md]: > most notably variance checks.

is this true?

File: website/docs/procs.md [hunk 3]
```diff
@@ -17,105 +18,147 @@ At present, all parameters are assumed to be required positional parameters. We
...
 
-This error can be fixed without modifying the method body to use `block.call`:
-It's possible to give the method a named block parameter like `&blk`, and
-`yield` will still work exactly as it did, invoking the block regardless of
-whether or not it was named. Once the block parameter has a name, it's in turn
-possible to associate that name with a type:
+This is valid Ruby, and Sorbet will accept this code too. Implicitly, Sorbet
+will know that the method might accept a block, but it will treat the block
+itself as `T.untyped`. In order to give a signature to this method, the block
+paramter will need a name:
```
Comment (connorshea) [inline website/docs/procs.md]:
```suggestion
parameter will need a name:
```

File: website/docs/procs.md [hunk 4]
```diff
@@ -17,105 +18,147 @@ At present, all parameters are assumed to be required positional parameters. We
...
+to connect the `yield` call to the `blk` parameter automatically.
+
 ## Modifying the self type
 
-While blocks are usually evaluated in the context in which they appear, it's
-possible to use the methods `instance_exec` and `instance_eval` to evaluate them
-in some other specified context, which changes the meaning of `self` in the body
-of that block. To convey this to Sorbet, we add a `.bind(...)` clause on the
-`T.proc` type that indicates the context in which the method will be invoked:
+Many Ruby constructs accept a block argument in one context, but then execute it
```
Comment (connorshea) [inline website/docs/procs.md]: should be `it in`?

**Pull request merged**